compiler/rustc_hir_typeck/src/op.rs

```diff
@@ -249,7 +249,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         );
 
         // see `NB` above
-        let rhs_ty = self.check_expr_coercible_to_type(rhs_expr, rhs_ty_var, Some(lhs_expr));
+        let rhs_ty = self.check_expr_coercible_to_type_or_error(
+            rhs_expr,
+            rhs_ty_var,
+            Some(lhs_expr),
+            |err| {
+                let ty = self.check_expr_with_hint(rhs_expr, rhs_ty_var);
```

Comment:
This is redundant because the same thing is already done in `check_expr_coercible_to_type_or_error()`. 

However, if we write `self.suggest_swapping_lhs_and_rhs(err, ty, lhs_expr, op);` directly into `check_expr_coercible_to_type_or_error()`, we have to add a parameter: `hir::BinOp` to `check_expr_coercible_to_type_or_error()`.

Sorry for that I don't know how to fix this :(

Reply:
you can add the type of the checked rhs expr as an argument to the `extend_err` closure.